browser-switch/src/main/java/com/braintreepayments/api/BrowserSwitchClient.java

```diff
@@ -119,25 +123,30 @@ private boolean isValidRequestCode(int requestCode) {
     /**
      * Completes the browser switch flow and returns a browser switch result if a match is found for
      * the given {@link BrowserSwitchRequest}
+     *
+     * @param intent         the intent to return to your application containing a deep link result from the
+     *                       browser flow
      * @param pendingRequest the {@link BrowserSwitchPendingRequest.Started} returned from
-     * {@link BrowserSwitchClient#start(ComponentActivity, BrowserSwitchOptions)}
-     * @param intent the intent to return to your application containing a deep link result from the
-     *               browser flow
-     * @return a {@link BrowserSwitchResult.Success} if the browser switch was successfully
-     * completed, or {@link BrowserSwitchResult.NoResult} if no result can be found for the given
-     * {@link BrowserSwitchPendingRequest.Started}. A {@link BrowserSwitchResult.NoResult} will be
+     *                       {@link BrowserSwitchClient#start(ComponentActivity, BrowserSwitchOptions)}
+     * @return a {@link BrowserSwitchStartResult.Success} if the browser switch was successfully
+     * completed, or {@link BrowserSwitchStartResult.NoResult} if no result can be found for the given
+     * {@link BrowserSwitchPendingRequest.Started}. A {@link BrowserSwitchStartResult.NoResult} will be
      * returned if the user returns to the app without completing the browser switch flow.
      */
-    public BrowserSwitchResult completeRequest(@NonNull BrowserSwitchPendingRequest.Started pendingRequest, @Nullable Intent intent) {
+    public BrowserSwitchCompleteRequestResult completeRequest(@NonNull Intent intent, @NonNull String pendingRequest) {
```

Comment:
I honestly think `BrowserSwitch.start()` and `BrowserSwitch.finish()` would work well for naming. The return object types could then be named `BrowserSwitchStartResult` / `BrowserSwitchFinishResult`. It's clear, concise and by convention this naming pattern maps the result objects to their respective SDK method invocation.

Reply:
Going to take this offline in a separate discussion.